src/Fixer/ClassNotation/OrderedInterfacesFixer.php

```diff
@@ -0,0 +1,221 @@
+    const OPTION_ORDER = 'order';
+
+    /** @internal */
+    const DIRECTION_ASCEND = 'ascend';
+
+    /** @internal */
+    const DIRECTION_DESCEND = 'descend';
+
+    /** @internal */
+    const ORDER_ALPHA = 'alpha';
+
+    /** @internal */
+    const ORDER_LENGTH = 'length';
+
+    /**
+     * Array of supported directions in configuration.
+     *
+     * @var string[]
+     */
+    private $supportedDirectionOptions = [
+        self::DIRECTION_ASCEND,
+        self::DIRECTION_DESCEND,
+    ];
+
+    /**
+     * Array of supported orders in configuration.
+     *
+     * @var string[]
+     */
+    private $supportedOrderOptions = [
+        self::ORDER_ALPHA,
+        self::ORDER_LENGTH,
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Orders the interfaces in an `implements` or `interface extends` clause.',
+            [
+                new CodeSample(
+                    "<?php\n\nfinal class ExampleA implements Gamma, Alpha, Beta {}\n\ninterface ExampleB extends Gamma, Alpha, Beta {}\n"
+                ),
+                new CodeSample(
+                    "<?php\n\nfinal class ExampleA implements Gamma, Alpha, Beta {}\n\ninterface ExampleB extends Gamma, Alpha, Beta {}\n",
+                    [self::OPTION_DIRECTION => self::DIRECTION_DESCEND]
+                ),
+                new CodeSample(
+                    "<?php\n\nfinal class ExampleA implements MuchLonger, Short, Longer {}\n\ninterface ExampleB extends MuchLonger, Short, Longer {}\n",
+                    [self::OPTION_ORDER => self::ORDER_LENGTH]
+                ),
+                new CodeSample(
+                    "<?php\n\nfinal class ExampleA implements MuchLonger, Short, Longer {}\n\ninterface ExampleB extends MuchLonger, Short, Longer {}\n",
+                    [
+                        self::OPTION_ORDER => self::ORDER_LENGTH,
+                        self::OPTION_DIRECTION => self::DIRECTION_DESCEND,
+                    ]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_IMPLEMENTS)
+            || $tokens->isAllTokenKindsFound([T_INTERFACE, T_EXTENDS]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_IMPLEMENTS)) {
+                if (!$token->isGivenKind(T_EXTENDS)) {
+                    continue;
+                }
+
+                $nameTokenIndex = $tokens->getPrevMeaningfulToken($index);
+                $interfaceTokenIndex = $tokens->getPrevMeaningfulToken($nameTokenIndex);
+                $interfaceToken = $tokens[$interfaceTokenIndex];
+
+                if (null === $interfaceToken || !$interfaceToken->isGivenKind(T_INTERFACE)) {
```

Comment:
I think `null === $interfaceToken` is never true here?

Reply:
@dmvdbrugge , can you answer this as well, please ?

Reply:
I see we don't start fixing files that are invalid to begin with, so no this will never be null. I'll remove the check.